code/__DEFINES/dcs/signals.dm

```diff
@@ -1003,6 +1003,10 @@
 ///from base of datum/component/two_handed/proc/unwield(mob/living/carbon/user): (/mob/user)
 #define COMSIG_TWOHANDED_UNWIELD "twohanded_unwield"
 
+// /datum/component/ice_cream_holder signals.
+/// Adds a serving of ice cream to the object. Bool. Refer to __DEFINES/food.dm for flavour defines: (flavour, reagents, custom_name, forced)
+#define COMSIG_ICE_CREAM_ADD_SERVING "ice_cream_add_serving"
```

Comment:
This seems completely redundant, it's only used to initialize the borg/crematorium types. And the signal is so specific it's not very useful.

Reply:
So do you suggest I use `GetComponent()` and handle it without signals instead?

Reply:
For this case i'd suggest creating presets for the robo/crematorium cones that start with the given flavours.